### PR TITLE
added pterodactyl config

### PR DESCRIPTION
**Pull request by GetLifty**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/reverse-proxy-confs/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description
Adds two new subdomain config files! One is for the pterodactyl game server panel, and the other is for pterodactyl nodes.

## Benefits of this PR and context
Pterodactyl panel reverse proxy is not very well documented outside of their own implementation. When trying to set it up myself I found only one guide designed for nginx proxy manager and had to adapt from that.

## How Has This Been Tested?
Tested on my personal setup. Works perfectly including API access, SFTP and other included features. Have also tested on a friends server to similar success.

## Source / References
N/A

### CODE REVIEW

File: pterodactylnode.subdomain.conf.sample [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        include /config/nginx/proxy.conf;
+        include /config/nginx/resolver.conf;
+        set $upstream_app pterodactylnode;
+        set $upstream_port 443;
+        set $upstream_proto http;
+        proxy_pass $upstream_proto://$upstream_app:$upstream_port/api/servers/$serverid/ws;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
+    }
```
Comment (nemchik) [inline pterodactylnode.subdomain.conf.sample:36]:
```suggestion
```
This section should go below the main location (for consistency with other confs) and could be a bit simpler. The proxy headers aren't needed (they are already included in the proxy.conf) and the regex could be simpler (covering all of the `/api` endpoint).

P.S. The whitespace (empty lines) might need cleaning up if you accept this suggestion as a commit.

File: pterodactylnode.subdomain.conf.sample [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        #include /config/nginx/authentik-location.conf;
+		
+        include /config/nginx/proxy.conf;
+        include /config/nginx/resolver.conf;
+        set $upstream_app pterodactylnode;
+        set $upstream_port 443;
+        set $upstream_proto http;
+        proxy_pass $upstream_proto://$upstream_app:$upstream_port;
+
+    }
```
Comment (nemchik) [inline pterodactylnode.subdomain.conf.sample:59]:
```suggestion
    }
    
    location ~ (/pterodactylnode)?/api {
        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
        set $upstream_app pterodactylnode;
        set $upstream_port 443;
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port/api/servers/$serverid/ws;

    }
```

File: pterodactylpanel.subdomain.conf.sample [hunk 3]
```diff
@@ -0,0 +1,48 @@
+## Version 2023/05/08
```
Comment (nemchik) [inline pterodactylpanel.subdomain.conf.sample:1]: Can you rename this to `pterodactyl.subdomain.conf.sample`

File: pterodactylnode.subdomain.conf.sample [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        #include /config/nginx/authelia-location.conf;
+
+        # enable for Authentik (requires authentik-server.conf in the server block)
+        #include /config/nginx/authentik-location.conf;
+		
+        include /config/nginx/proxy.conf;
+        include /config/nginx/resolver.conf;
+        set $upstream_app pterodactylnode;
+        set $upstream_port 443;
+        set $upstream_proto http;
```
Comment (nemchik) [inline pterodactylnode.subdomain.conf.sample:56]: This is more of a question (just to confirm) is this meant to work on `443` using `http`? Or is it meant to be `https`?

**Review by nemchik — CHANGES_REQUESTED** [review]

File: pterodactylnode.subdomain.conf.sample (hunk 4 above)
Comment (GetLifty) [reply pterodactylnode.subdomain.conf.sample:56]: It is meant to be http. The daemon port for the node is user selectable, but when I was researching setting my server up, it seemed the general consensus was to use 8443 or 443 (despite the fact it isnt https).

File: pterodactylpanel.subdomain.conf.sample (hunk 3 above)
Comment (GetLifty) [reply pterodactylpanel.subdomain.conf.sample]: Done

**Comment by nemchik** [comment]
https://github.com/linuxserver/reverse-proxy-confs/pull/571/commits/8355cdc795ef6cf74b9ba5f7d67e65928e001a47 this isn't what I meant about whitespace. Please keep 4 space indentation (not tabs). What I meant was my suggestion may have left extra empty lines as I did not account for the empty lines when highlighting. If there's no double empty lines then it's fine.

**Comment by GetLifty** [comment]
> [8355cdc](https://github.com/linuxserver/reverse-proxy-confs/commit/8355cdc795ef6cf74b9ba5f7d67e65928e001a47) this isn't what I meant about whitespace. Please keep 4 space indentation (not tabs). What I meant was my suggestion may have left extra empty lines as I did not account for the empty lines when highlighting. If there's no double empty lines then it's fine.

Sorry, if you cant tell I am pretty new to github (and generally programming). My tired brain assumed that the indentation was causing the check failure and manually tried to fix it :(.

**Comment by GetLifty** [comment]
Ok I reverted the commit (after spending embarrasingly long figuring out how). Still not 100% on the check failure but if its something I am missing feel free to let me know. Ive not got any more time to work today so I will likely fix this tommorow.

File: pterodactylnode.subdomain.conf.sample [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    }
+    
+    location ~ (/pterodactylnode)?/api {
+        include /config/nginx/proxy.conf;
+        include /config/nginx/resolver.conf;
+        set $upstream_app pterodactylnode;
+        set $upstream_port 443;
+        set $upstream_proto http;
+        proxy_pass $upstream_proto://$upstream_app:$upstream_port/api/servers/$serverid/ws;
```
Comment (nemchik) [inline pterodactylnode.subdomain.conf.sample:56]:
```suggestion
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;
```

**Review by nemchik — CHANGES_REQUESTED** [review]
This should fix the failing action check.

**Review by nemchik — APPROVED** [review]

**Pull request merged**